core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1278,11 +1287,22 @@ private[spark] object SparkSubmitUtils {
       remoteRepos: Option[String],
       ivyPath: Option[String]): IvySettings = {
     val file = new File(settingsFile)
-    require(file.exists(), s"Ivy settings file $file does not exist")
-    require(file.isFile(), s"Ivy settings file $file is not a normal file")
+    // When running driver in cluster mode, the settingsFile is localized and so needs to be
+    // accessed using just the file name and not the full file path
+    val localizedFile = new File(file.getName)
+    val resolvedFile = if (file.exists()) {
+      Some(file)
+    } else if (localizedFile.exists()) {
```

Comment:
Doesn't it pick the file in the current working directory? It wouldn't make sense if you're running it on non-cluster modes.

Reply:
Yeah, agreed. I have elaborated on the difficulty to implement this cleanly in https://github.com/apache/spark/pull/31591#issuecomment-783969152